File: 05_hash_tables/java/02_check_voter.java
import java.util.*;
import java.lang.*;
import java.io.*;


class grokkingAlgorithms
{	
    public static HashMap<String, Boolean> voted = new HashMap<String, Boolean>();

    public static void check_voter(String name){
        if (voted.containsKey(name)){
            System.out.println("kick them out!");
        } else {
            voted.put(name, true);
            System.out.println("let them vote!");
        }
    }

	public static void main (String[] args) throws java.lang.Exception
	{
        check_voter("tom");
        check_voter("mike");
        check_voter("mike");
	}
}

File: 05_hash_tables/javascript/02_check_voter.js
var voted = new Map();
function check_voter(name){
    if(voted[name] == true){
        console.log("kick them out!");
    }
    else{
        voted[name] = true;
        console.log("let them vote!");
    }
}

check_voter("tom");
check_voter("mike");
check_voter("mike");
